nemo/collections/asr/models/msdd_models.py

```diff
@@ -1408,8 +1417,114 @@ def run_overlap_aware_eval(
                 out_rttm_dir=self.out_rttm_dir,
             )
             output = score_labels(
-                self.AUDIO_RTTM_MAP, all_reference, all_hypothesis, collar=collar, ignore_overlap=ignore_overlap,
+                rttm_map, all_reference, all_hypothesis, collar=collar, ignore_overlap=ignore_overlap,
             )
             outputs.append(output)
         logging.info(f"  \n")
         return outputs
+
+    @classmethod
+    def from_pretrained(
+        cls,
+        diar_model_name: str = 'diar_msdd_telephonic',
+        vad_model_name: str = 'vad_multilingual_marblenet',
+        speaker_model_name: str = 'titanet_large',
+        domain: Optional[str] = None,
+        device: Union[torch.device, str] = "auto",
+    ):
+        """
+        Instantiate a `NeuralDiarizer` to run Speaker Diarization.
+
+        Args:
+            diar_model_name (str): Path/Name of the neural diarization model to load.
+            vad_model_name (str): Path/Name of the voice activity detection (VAD) model to load.
+            speaker_model_name (str): Path/Name of the speaker model to load.
+            domain (str): Specific domain of thea udio, supports "telephonic" and "meeting" (default is general).
+            device (torch.device): Device we are running on. "auto selects CUDA if available ('cpu', 'cuda').
+
+        Returns:
+            `NeuralDiarizer`
+        """
+        cfg = NeuralInferenceConfig.from_domain(domain, device)
+        cfg.diarizer.msdd_model.model_path = diar_model_name
+        cfg.diarizer.vad.model_path = diar_model_name
+        cfg.diarizer.speaker_embeddings.model_path = diar_model_name
+        return cls(cfg)
+
+    def __call__(
+        self,
+        inference_input: Union[str, List[str]],
```

Comment:
thats quite an ugly name; how about just calling it `input_path`? I went for `inference_input` in case we wanted to just support loading a manifest_filepath.

Reply:
Path2audio_files will be deprecated with future release anyway, it's fine to change the name here. But not generic input_path - audio_filepaths, audio_paths or some variation should be used. If more types of data are used, then split the args

Reply:
`audio_filepath` ? since we do inference of only one file now.